java/ql/test/library-tests/frameworks/spring/util/Test.java

```diff
@@ -0,0 +1,1445 @@
+		{
+			// "org.springframework.util;MultiValueMap;true;addAll;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			MultiValueMap<Object, Object> mvm = null;
+			mvm.set(source(), "someValue");
+			out.addAll(mvm);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;addIfAbsent;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.addIfAbsent(in, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;addIfAbsent;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.addIfAbsent(in, (Object)null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;addIfAbsent;;;Argument[1];Element of MapValue of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.addIfAbsent(null, in);
+			sink(getElement((List)getMapValue(out))); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;addIfAbsent;;;Argument[1];Element of MapValue of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.addIfAbsent((Object)null, in);
+			sink(getElement((List)getMapValue(out))); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;getFirst;;;Element of MapValue of Argument[-1];ReturnValue;value"
+			Object out = null;
+			MultiValueMapAdapter in = new MultiValueMapAdapter(Map.of(null, List.of(source())));
+			out = in.getFirst(null);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;getFirst;;;Element of MapValue of Argument[-1];ReturnValue;value"
+			Object out = null;
+			MultiValueMap in = (MultiValueMap)Map.of(null, List.of(source()));
+			out = in.getFirst(null);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;getFirst;;;Element of MapValue of Argument[-1];ReturnValue;value"
+			Object out = null;
+			MultiValueMap in = (MultiValueMap)Map.of(null, List.of(source()));
+			out = in.getFirst((Object)null);
+			sink(out); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;set;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMapAdapter out = null;
+			Object in = (Object)source();
+			out.set(in, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;set;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.set(in, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.springframework.util;MultiValueMap;true;set;;;Argument[0];MapKey of Argument[-1];value"
+			MultiValueMap out = null;
+			Object in = (Object)source();
+			out.set(in, (Object)null);
+			sink(getMapKey(out)); // $hasValueFlow
```

Comment:
These three look like duplicates of each other?

Reply:
@smowton Is this something that can be traced back to a bug in the test generation?

Reply:
Two differ (the top one is checking inheritance by `MultiValueMapAdapter`); dropped the last one.